examples/platform/silabs/SoftwareFaultReports.cpp

```diff
@@ -131,24 +135,59 @@ extern "C" __attribute__((used)) void debugHardfault(uint32_t * sp)
     ChipLogError(NotSpecified, "LR          0x%08lx", lr);
     ChipLogError(NotSpecified, "PC          0x%08lx", pc);
     ChipLogError(NotSpecified, "PSR         0x%08lx", psr);
+#if SILABS_LOG_OUT_UART
+    uartFlushTxQueue();
+#endif // SILABS_LOG_OUT_UART
 #endif // SILABS_LOG_ENABLED
 
     configASSERTNULL(NULL);
 }
 
 /**
- * Override default hard-fault handler
+ * Log a fault to the debugHardfault function.
+ * This function is called by the fault handlers to log the fault details.
  */
+extern "C" __attribute__((naked)) void LogFault_Handler(void)
+{
+    uint32_t * sp;
+    __asm volatile("tst lr, #4 \n"
+                   "ite eq \n"
+                   "mrseq %0, msp \n"
+                   "mrsne %0, psp \n"
+                   : "=r"(sp));
+    debugHardfault(sp);
+}
+
 #ifndef SL_CATALOG_ZIGBEE_STACK_COMMON_PRESENT
+extern "C" __attribute__((naked)) void NMI_Handler(void)
+{
+    __asm volatile("b LogFault_Handler");
```

Comment:
why does it need to be an assembly branch operation? Why can we not just call ` LogFault_Handler();`

Reply:
So my compiler knowledge is  bit rusty, but I think  that a C call would have the compiler would push the return address of LogFault_Handler on the callstack, which is not relevant debugging info. 

I don't think this would be dramatic either but I would have to test it cannot generate any other corruption of register/stack pointer with different optimization, so I felt it was safer to just have the assembly branch that doesn't alter anything else

Reply:
I'm sure there are also proper guards that I could put around a C call to make this behave as I want as well, so I can also look into it if we really want to get rid of this assembly branch instruction